### ISSUE TITLE
Conflict with LND other port 9735

### ISSUE THREAD

**Issue by BobleChinois** [btcpayserver/btcpayserver-docker#17]
# Summary
I want to start a btcpayserver docker to experiment on the testnet. I already run a bitcoind and LND instance on the mainnet on the same machine (a small home server on Ubuntu).

I followed instructions on the README page, and ran `. ./btcpay-setup.sh -i`, everything was ok, except an error that kept c-lightning from starting.

Here is the exact error message : 
`ERROR: for clightning_bitcoin  Cannot start service clightning_bitcoin: driver failed programming external connectivity on endpoint btcpayserver_clightning_bitcoin (fdb8db643d2aae0fbc6bbd906c7fd97ed074a5ed2be74e887191293fbb5e4d56): Error starting userland proxy: listen tcp 0.0.0.0:9735: bind: address already in use`

`lsof -i -P -n | grep LISTEN` indeed shows `lnd      13495         bitcoin    8u  IPv6 511515      0t0  TCP *:9735 (LISTEN)`

Is there a way to change the port c-lightning is listening on setup ? Especially since my LND is running on main and this c-lighting will run on testnet.

## Environment setup
```
Description:    Ubuntu 18.04 LTS
Release:        18.04
4.15.0-23-generic
```
btcpayserver-docker commit `511db652e8228f566b87620fe94a8d00f162d120`

Thanks !

**Comment by NicolasDorier** [comment]
@BobleChinois yes, edit your `$BTCPAY_DOCKER_COMPOSE`, and restart btcpay with `btcpay-up.sh`.

However, keep in mind that at the next update (`btcpay-update.sh`), your changes will be overridden. I am open for PR for finding a solution which would allow you to keep your changes.

**Comment by BobleChinois** [comment]
@NicolasDorier Ok I'll try that, as for the way to avoid my change being overridden by update, I'll think about the problem.

**Comment by BobleChinois** [comment]
Ok back to this issue, I changed the port number in the docker compose file as you suggested, here is a screenshot of my current file : 
![btcpay-dockercompose](https://user-images.githubusercontent.com/11726747/43903554-036c33a8-9bed-11e8-9552-632d552bec81.png)
It's working, but I have a question. I noticed that `getinfo` will return this : 
```
{
  "id": "0307edc72b072239427baa54324c43a52c2edae2a1a2de81404787243cf864dfbe", 
  "port": 9735, 
  "address": [
    {
      "type": "ipv4", 
      "address": "5.51.82.169", 
      "port": 9730
    }
  ], 
  "binding": [
    {
      "type": "ipv4", 
      "address": "0.0.0.0", 
      "port": 9730
    }
  ], 
  "version": "basedon-aeafe4dbe7d5c61f664c18417698866b0d70252f-modded", 
  "blockheight": 1382421, 
  "network": "testnet"
```
Is it normal that "port" still indicate "9735" ?

**Comment by NicolasDorier** [comment]
Please, do not change the internal port. Only the external. BTCPay is using the internal one in its config.

I would say: This is the internal port, so this is normal. But since you modified the internal port (and you should not), this is not normal. But because something is not normal about something you should have not have done, things happen to work...

TL;DR: Keep the internal port as it was in the configuration file.

**Comment by BobleChinois** [comment]
Hum, it brought us back to the initial issue I'm afraid. Here is what I get when I run `./btcpay-up.sh` : 
```
ERROR: for btcpayserver_clightning_bitcoin  Cannot start service clightning_bitcoin: driver failed programming external connectivity on endpoint btcpayserver_clightning_bitcoin (259c875371f2f4aca3690c53f06e39650a1b906c995843ef8086540171125bCreating generated_nbxplorer_1           ... done
dy in use

ERROR: for clightning_bitcoin  Cannot start service clightning_bitcoin: driver failed programming external connectivity on endpoint btcpayserver_clightning_bitcoin (259c875371f2f4aca3690c53f06e39650a1b906c995843ef8086540171125ba9): Error starting userland proxy: listen tcp 0.0.0.0:9735: bind: address already in use
ERROR: Encountered errors while bringing up the project.

```

**Comment by NicolasDorier** [comment]
ah sorry, in `ports`, it is `external:internal`  so you need to only modify external
